themes/src/main/resources/theme/base/login/login-config-totp.ftl

```diff
@@ -18,7 +18,7 @@
             <#if mode?? && mode = "manual">
                 <li>
                     <p>${msg("loginTotpManualStep2")}</p>
-                    <p><span id="kc-totp-secret-key">${totp.totpSecretEncoded}</span></p>
+                    <p><span id="kc-totp-secret-key">${totpSecretEncoded}</span></p>
```

Comment:
Minor: Is it possible to revert this line (as well as the line 42) and keep using the `totpSecretEncoded` and `totpSecretQrCode` from the TotpBean? This may mean some slight refactoring of `org.keycloak.forms.login.freemarker.model.TotpBean` to read the stuff from authSession instead of what is currently used here.

Also the `totpSecret` field won't be needed on `TotpBean` with your latest changes and can be hopefully cleaned?

Also it will allow to few calls of `setAttribute` in the `UpdateTotp` class as attributes won't be needed to be set directly in the context.

Reply:
Changed to the suggested approach